packages/web/components/bridge/amount-screen.tsx

```diff
@@ -371,27 +371,37 @@ export const AmountScreen = observer(
           ) as Extract<SupportedAsset, { chainType: "cosmos" }>[],
           userCosmosAddress: fromCosmosCounterpartyAccount?.address,
         };
-      } else {
-        // TODO: can input solana or btc assets
+      } else if (fromChain?.chainType === "bitcoin" && supportedSourceAssets) {
         return {
-          type: fromChain?.chainType ?? "bitcoin",
-          assets: (supportedSourceAssets?.filter(
+          type: "bitcoin" as const,
+          assets: supportedSourceAssets.filter(
             (asset) => asset.chainType === "bitcoin"
-          ) ?? []) as Extract<SupportedAsset, { chainType: "bitcoin" }>[],
+          ) as Extract<SupportedAsset, { chainType: "bitcoin" }>[],
+        };
+      } else {
+        // solana
+        return {
+          type: "solana" as const,
+          assets: (supportedSourceAssets ?? []).filter(
+            (asset) => asset.chainType === "solana"
+          ) as Extract<SupportedAsset, { chainType: "solana" }>[],
         };
       }
     })();
     const { data: assetsBalances, isLoading: isLoadingAssetsBalance } =
       api.local.bridgeTransfer.getSupportedAssetsBalances.useQuery(
         { source: balanceSource },
         {
-          enabled: !isNil(fromChain) && !isNil(supportedSourceAssets),
+          enabled:
+            !isNil(fromChain) &&
+            !isNil(supportedSourceAssets) &&
+            supportedSourceAssets.length > 0,
 
           select: (data) => {
             let nextData: typeof data = data;
 
             // Filter out assets with no balance
-            if (nextData) {
+            if (nextData && nextData.length) {
```

Comment:
This shouldn't happen, but we should check that the balances query returned is not empty before using `[0]`